x-pack/plugins/code/server/plugin.ts

```diff
@@ -73,13 +78,17 @@ export class CodePlugin {
       },
     });
 
-    return deepFreeze({
```

Comment:
what is the problem to make it immutable?

Reply:
both the `http.createRouter()` and `elasticsearch.adminClient` could be the problem when call some functions on them.